var/spack/repos/builtin/packages/foam-extend/package.py

```diff
@@ -114,187 +108,73 @@ class FoamExtend(Package):
     depends_on('parmgridgen',      when='+parmgridgen')
     depends_on('paraview@:5.0.1',  when='+paraview')
 
-    # Some user settings, to be adjusted manually or via variants
-    foam_cfg = {
-        'WM_COMPILER':        'Gcc',  # <- %compiler
-        'WM_ARCH_OPTION':     '64',   # (32/64-bit on x86_64)
-        # FUTURE? 'WM_LABEL_SIZE':      '32',  # <- +int64
-        'WM_PRECISION_OPTION': 'DP',  # <- +float32
-        'WM_COMPILE_OPTION':  'SPACKOpt',   # Do not change
-        'WM_MPLIB':           'USER',       # USER | USERMPI
-    }
+    # General patches
+    assets = ['spack-Allwmake', 'README-spack']
 
-    # The system description is frequently needed
-    foam_sys = {
-        'WM_ARCH':      None,
-        'WM_COMPILER':  None,
-        'WM_OPTIONS':   None,
+    # Some user config settings
+    config = {
+        'label-size': False,    # <- No int32/int64 support
+        'mplib': 'USERMPI',     # USER | USERMPI
     }
 
+    # The openfoam architecture, compiler information etc
+    _foam_arch = None
+
     # Content for etc/prefs.{csh,sh}
     etc_prefs = {}
 
     # Content for etc/config.{csh,sh}/ files
     etc_config = {}
 
-    build_script = './spack-Allwmake'  # <- Generated by patch() method.
-    # phases = ['configure', 'build', 'install']
-    # build_system_class = 'OpenfoamCom'
+    build_script = 'spack-Allwmake'  # <- Generated by patch() method.
+
+    phases = ['configure', 'build', 'install']
+
+    #
+    # - End of definitions / setup -
+    #
 
     def setup_environment(self, spack_env, run_env):
-        run_env.set('FOAM_INST_DIR', self.prefix)
+        run_env.set('FOAM_INST_DIR', os.path.dirname(self.projectdir)),
+        run_env.set('FOAM_PROJECT_DIR', self.projectdir)
         run_env.set('WM_PROJECT_DIR', self.projectdir)
-
-    @property
-    def _canonical(self):
-        """Canonical name for this package and version"""
-        return 'foam-extend-{0}'.format(self.version.up_to(2))
+        for d in ['wmake', self.archbin]:  # bin already added automatically
+            run_env.prepend_path('PATH', join_path(self.projectdir, d))
+        run_env.set('MPI_BUFFER_SIZE', "20000000")
```

Comment:
This is always set by the OpenFOAM environment scripts and seems to apply to all MPIs except SGI. Presumably this could be addressed if/when the setup_environment gets sorted.